mzLib/Omics/Digestion/LysisProduct.cs

```diff
@@ -0,0 +1,192 @@
+        public int OneBasedStartResidue { get; }// the residue number at which the peptide begins (the first residue in a protein is 1)
+        public int OneBasedEndResidue { get; }// the residue number at which the peptide ends
+        public int MissedCleavages { get; } // the number of missed cleavages this peptide has with respect to the digesting protease
+        public virtual char PreviousResidue => OneBasedStartResidue > 1 ? Parent[OneBasedStartResidue - 2] : '-';
+
+        public virtual char NextResidue => OneBasedEndResidue < Parent.Length ? Parent[OneBasedEndResidue] : '-';
+        public string BaseSequence =>
+            _baseSequence ??= Parent.BaseSequence.Substring(OneBasedStartResidue - 1,
+                OneBasedEndResidue - OneBasedStartResidue + 1);
+        public CleavageSpecificity CleavageSpecificityForFdrCategory { get; set; } //structured explanation of source
+        public int Length => BaseSequence.Length; //how many residues long the peptide is
+        public char this[int zeroBasedIndex] => BaseSequence[zeroBasedIndex];
+
+        protected static IEnumerable<Dictionary<int, Modification>> GetVariableModificationPatterns(Dictionary<int, List<Modification>> possibleVariableModifications, int maxModsForPeptide, int peptideLength)
+        {
+            if (possibleVariableModifications.Count == 0)
+            {
+                yield return null;
+            }
+            else
+            {
+                var possible_variable_modifications = new Dictionary<int, List<Modification>>(possibleVariableModifications);
+
+                int[] base_variable_modification_pattern = new int[peptideLength + 4];
+                var totalAvailableMods = possible_variable_modifications.Sum(b => b.Value == null ? 0 : b.Value.Count);
+                for (int variable_modifications = 0; variable_modifications <= Math.Min(totalAvailableMods, maxModsForPeptide); variable_modifications++)
+                {
+                    foreach (int[] variable_modification_pattern in GetVariableModificationPatterns(new List<KeyValuePair<int, List<Modification>>>(possible_variable_modifications),
+                        possible_variable_modifications.Count - variable_modifications, base_variable_modification_pattern, 0))
+                    {
+                        yield return GetNewVariableModificationPattern(variable_modification_pattern, possible_variable_modifications);
+                    }
+                }
+            }
+        }
+
+        protected Dictionary<int, Modification> GetFixedModsOneIsNterminusOrFivePrime(int peptideLength,
+            IEnumerable<Modification> allKnownFixedModifications)
+        {
+            var fixedModsOneIsNterminus = new Dictionary<int, Modification>(peptideLength + 3);
+            foreach (Modification mod in allKnownFixedModifications)
+            {
+                switch (mod.LocationRestriction)
+                {
+                    case "5'-terminal.":
```

Comment:
terminus not terminal?

Reply:
This maintains naming convention from terminal mods from proteomics